providers/proxy.rb

```diff
@@ -0,0 +1,200 @@
+
+      		gecos_ws_mgmt_desktop_settings "proxy config https port" do
+	      		provider "gecos_ws_mgmt_gsettings"
+        		schema "org.gnome.system.proxy.https"
+        		type "string"
+        		username username
+						name "port"
+        		value proxy_settings['https_proxy_port']
+      		end.run_action(:set)
+				
+					# ENVIROMENT VARIABLES HTTP_PROXY, HTTPS_PROXY
+	        ruby_block "include-bashrc-user" do
+	          block do
+  	          sed = Chef::Util::FileEdit.new("/home/#{username}/.bashrc")
+							sed.search_file_replace_line(/HTTP_PROXY/i,"export HTTP_PROXY=\"#{proxy_settings['http_proxy']}:#{proxy_settings['http_proxy_port']}\"")
+      	      #sed.write_file
+							sed.search_file_replace_line(/HTTPS_PROXY/i,"export HTTPS_PROXY=\"#{proxy_settings['https_proxy']}:#{proxy_settings['https_proxy_port']}\"")
+      	      #sed.write_file
+    	        sed.insert_line_if_no_match(/HTTP_PROXY/i,"export HTTP_PROXY=\"#{proxy_settings['http_proxy']}:#{proxy_settings['http_proxy_port']}\"")
+      	      sed.write_file
+							sed.insert_line_if_no_match(/HTTPS_PROXY/i,"export HTTPS_PROXY=\"#{proxy_settings['https_proxy']}:#{proxy_settings['https_proxy_port']}\"")
+      	      sed.write_file
+        	  end
+          	action :nothing
+        	end.run_action(:run)
+
+
+				elsif !proxy_settings['disable_proxy'] and proxy_settings['proxy_autoconfig_url'] != nil
+      		gecos_ws_mgmt_desktop_settings "proxy config autoconfig mode" do
+	      		provider "gecos_ws_mgmt_gsettings"
+        		schema "org.gnome.system.proxy"
+        		type "string"
+        		username username
+						name "mode"
+        		value "auto"
+      		end.run_action(:set)
+				
+      		gecos_ws_mgmt_desktop_settings "proxy config autoconfig url" do
+	      		provider "gecos_ws_mgmt_gsettings"
+        		schema "org.gnome.system.proxy"
+        		type "string"
+        		username username
+						name "autoconfig-url"
+        		value proxy_settings['proxy_autoconfig_url']
+      		end.run_action(:set)
+			
+				elsif proxy_settings['disable_proxy']
+
+          gecos_ws_mgmt_desktop_settings "proxy config None" do
+            provider "gecos_ws_mgmt_gsettings"
+            schema "org.gnome.system.proxy"
+            type "string"
+            username username
+            name "mode"
+            value "none"
+          end.run_action(:set)
+
+				 	ruby_block "include-bashrc-user" do
+	       		block do
+          		sed = Chef::Util::FileEdit.new("/home/#{username}/.bashrc")
+            	sed.search_file_delete_line(/HTTPS?_PROXY/i)
+              sed.write_file
+          	end
+          	action :nothing
+        	end.run_action(:run)
+
+				end
+
+				# FIREFOX
+				gecos_ws_mgmt_appconfig_firefox "Firefox proxy configuration" do
+	      	provider "gecos_ws_mgmt_appconfig_firefox"
+				 config_firefox proxy_settings
+					job_ids new_resource.job_ids
+					support_os new_resource.support_os
+				end.run_action(:setup)
+				
+				# THUNDERBIRD
```

Comment:
This will configure system level proxy for thunderbird, not user level proxy. It will do it several times if the policy applies to several users of this workstation, since it's inside the users loop.